fuzzing/disco_fuzzer.go

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package fuzzing
+
+import "tailscale.com/disco"
+
+func FuzzDiscoParser(data []byte) int {
+	_, err := disco.Parse(data)
+	if err != nil {
+		return 0
+	}
```

Comment:
Once we have a valid `disco.Message`, we should fuzz using that as well. E.g. we should check that AppendMessage round trips: Use AppendMessage to encode the Message, then parse it again, and then use reflect.DeepEqual to confirm that the twice-parsed message is identical to the once-parsed message.